test/jtreg-ext/requires/VMProps.java

```diff
@@ -658,14 +658,14 @@ private String jdkContainerized() {
      * Checks if we are in <i>almost</i> out-of-box configuration, i.e. the flags
      * which JVM is started with don't affect its behavior "significantly".
      * {@code TEST_VM_FLAGLESS} enviroment variable can be used to force this
-     * method to return true and allow any flags.
+     * method to return true or false and allow or reject any flags.
      *
      * @return true if there are no JVM flags
      */
     private String isFlagless() {
         boolean result = true;
         if (System.getenv("TEST_VM_FLAGLESS") != null) {
-            return "" + result;
+            return "" + "true".equalsIgnoreCase(System.getenv("TEST_VM_FLAGLESS"));
```

Comment:
I would slightly prefer extracting the `System.getenv()` call into a local and reusing it, but this is fine as well.